projects/core/src/cms/facade/page-meta.service.ts

```diff
@@ -1,20 +1,37 @@
 import { Inject, Injectable, Optional } from '@angular/core';
 import { combineLatest, Observable, of } from 'rxjs';
-import { debounceTime, filter, map, switchMap } from 'rxjs/operators';
+import {
+  debounceTime,
+  filter,
+  map,
+  shareReplay,
+  switchMap,
+} from 'rxjs/operators';
 import { resolveApplicable } from '../../util/applicable';
 import { Page, PageMeta } from '../model/page.model';
 import { PageMetaResolver } from '../page/page-meta.resolver';
 import { CmsService } from './cms.service';
+import { UnifiedInjector } from '../../lazy-loading/unified-injector';
 
 @Injectable({
   providedIn: 'root',
 })
 export class PageMetaService {
+  private resolvers$: Observable<PageMetaResolver[]> = this.unifiedInjector
+    ? (this.unifiedInjector
+        .getMulti(PageMetaResolver)
+        .pipe(shareReplay({ bufferSize: 1, refCount: true })) as Observable<
+        PageMetaResolver[]
+      >)
+    : of(this.resolvers);
+
   constructor(
     @Optional()
     @Inject(PageMetaResolver)
     protected resolvers: PageMetaResolver[],
-    protected cms: CmsService
+    protected cms: CmsService,
+    @Optional()
+    protected unifiedInjector?: UnifiedInjector
```

Comment:
I'm thinking to keep it that way, as for cases where the user doesn't use lazy loading at all, it can still safely use original constructor, what do you think?